File: js/task3a/S3_OBJ_06.js
function groupBy(items, key) {
  const result = items.reduce((acc, item) => {
    const value = item[key];

    if (!acc[value]) {
      acc[value] = [];
    }

    acc[value].push(item);
    return acc;
  }, {});
  console.log(result); 
  return result;
}
const data = [
  { type: 'fruit', name: 'apple' },
  { type: 'fruit', name: 'banana' },
  { type: 'vegetable', name: 'carrot' },
];

groupBy(data, 'type');

File: python/task3a/S3_OBJ_06.py
def groupBy(items, key):
  result = {}
  for item in items:
      value = item.get(key)
      if value not in result:
          result[value] = []
      result[value].append(item)
  print(result)
  return result


data = [
  {"type": "fruit", "name": "apple"},
  {"type": "fruit", "name": "banana"},
  {"type": "vegetable", "name": "carrot"},
]

groupBy(data, "type")
